controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -121,9 +121,6 @@ func (r *KubeadmControlPlaneReconciler) SetupWithManager(ctx context.Context, mg
 	r.ssaCache = ssa.NewCache()
 
 	if r.managementCluster == nil {
-		if r.ClusterCache == nil {
```

Comment:
Just a minor cleanup, this check is never used because we would already panic above now.

I have a follow-up to introduce proper input validation for all SetupWithManager funcs